### PR TITLE
Allow to set Zero value on numeric preference

### LINKED ISSUE #8523: Zero value not saved in numeric Preference
### Bug description

When an extension defines a numeric preference, for which the zero (0) value is possible, this value is not saved correctly: if the user sets the value as 0, the value is changed to the default value of the preference.

https://github.com/user-attachments/assets/18b7d34d-edcb-4ea6-8850-27764f73b078

### Operating system

mac

### Installation Method

Installer from website/GitHub releases

### Version

next (development version)

### Steps to reproduce

_No response_

### Relevant log output

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by feloy**
Signed-off-by: Philippe Martin <phmartin@redhat.com>

### What does this PR do?

Allow to set Zero value on numeric preference

### What issues does this PR fix or reference?

Fixes #8523 

### How to test this PR?

- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

File: packages/renderer/src/lib/preferences/item-formats/NumberItem.spec.ts [hunk 1]
```diff
@@ -126,3 +130,25 @@ test('Expect decrement button only works one if minimum value is reached after o
...
+    maximum: 34,
+  };
+  const onChange = vi.fn();
+  const value = 1;
+  render(NumberItem, { record, value, onChange });
+
+  const input = screen.getByLabelText('decrement');
+  expect(input).toBeInTheDocument();
+  await userEvent.click(input);
+  vi.advanceTimersByTime(510);
```
Comment (benoitf) [inline packages/renderer/src/lib/preferences/item-formats/NumberItem.spec.ts:152]: should it use timers or just ticks ?

Comment (feloy) [reply packages/renderer/src/lib/preferences/item-formats/NumberItem.spec.ts:152]: I think that we need to rely on timers, as the Svelte component is not refreshed, but only a call to onChange is called after 500ms:

```
    valueUpdateTimeout = setTimeout(() => {
      onChange(recordId, newValue);
      lastValue = newValue;
    }, 500);
```

Comment (benoitf) [reply packages/renderer/src/lib/preferences/item-formats/NumberItem.spec.ts:152]: understood

**Review by benoitf — APPROVED** [review]

**Review by lstocchi — APPROVED** [review]
LGTM

**Pull request merged**